LaTeX/LaTeX.sublime-syntax

```diff
@@ -112,23 +112,29 @@ contexts:
     - include: scope:text.tex#macros
     - match: |-
         (?x)
+        # The defining command
         (
           (\\)
           (?:(?:new|renew|provide)command\*?)
         )
+        \s*
+        # the command to be defined, possibly surrounded by braces
         (?:
-          (\{)(\\[A-Za-z@]+)(\})
-          | (\\[A-Za-z@])+
+          (\{)\s*(\\[A-Za-z@]+)\s*(\})
+          | (\\[A-Za-z@]+)
         )
-        (?:(\[)(?:[^\]]*)(\]))*
+        \s*
+        # optional argument specification in square brackets
+        (?:(\[)(?:[^\]]*)(\])\s*)*
```

Comment:
The final `*` operator says there are possibly multiple bracket pairs. ST will only capture and scope one of them.

![grafik](https://user-images.githubusercontent.com/16542113/197380434-1160c8b5-7351-4110-82cb-eb03a80c91d2.png)

If the max amount of bracket pairs is known it may be enough to add some capture groups for each. 

A more modern approach however would be to refactor the pattern to push contexts for each part on stack. It would also improve UX by handling incomplete expressions better.

Reply:
To keep this changelist small, I'd go with the simple approach for now. LaTeX allows at most two optional arguments, 
the first specifying how many arguments the new command should take, and the second giving a default value for the first argument. 
What should be the scopes for these? 
`meta.function.parameters.default-value.python` for the default value, I guess?

Reply:
In the long run, I think it really would be necessary to split this into several contexts. The current approach, for example, also fails in situations like
```LaTeX
\newcommand\baz[1][dificu{]}t] {\textbf{#1}}
```
where the default value contains a closing bracket, escaped within a brace group.